### PR TITLE
Added pytest workflow for automatic testing

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Added pytest workflow pr. #13 .

This workflow runs the tests on Python 3.7, 3.8, 3.9. On the latest versions of Mac, Linux, and Windows. It also generates a comment to which contains the code coverage.

### CODE REVIEW

File: litterature.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Litererature
+# Littererature
```
Comment (saattrupdan) [inline litterature.md:1]: Typo: Literature

File: tests/datasets_test.py [hunk 2]
```diff
@@ -38,6 +41,7 @@ def test_load_reddit():
     assert isinstance(ds[0], dict)
     assert "text" in ds[0]
```
Comment (saattrupdan) [inline tests/datasets_test.py:43]: PEP8: Needs two empty lines between global functions.

File: tests/datasets_test.py [hunk 3]
```diff
@@ -12,6 +13,7 @@ def test_load_tweets():
     assert isinstance(tweet[0], dict)
     assert "text" in tweet[0]
```
Comment (saattrupdan) [inline tests/datasets_test.py:15]: PEP8: Needs two empty lines between global functions.

File: litterature.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
 This is a working document for literature. This is much for sharing and organizing meterial as opposed to the Zotero which is for citation management.
```
Comment (saattrupdan) [inline litterature.md:2]: Typo: material

File: tests/datasets_test.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from dfm.data.load import load_tweets, load_news, load_dagw, load_reddit, load_tokenizer_ds
 from datasets import IterableDataset, Dataset
 
+@pytest.mark.skip(reason="Dataset not publicly available")
```
Comment (saattrupdan) [inline tests/datasets_test.py:7]: PEP8: Needs two empty lines before global functions.

**Review by saattrupdan — CHANGES_REQUESTED** [review]
I'm guessing the files in the `dev` folder are auto-generated?

**Comment by KennethEnevoldsen** [comment]
No the files in dev are manually added by me (copy-paste from other projects). You could essentially wrap it in a GitHub action and remove it from the codebase (which I suppose would be ideal, but would have to spend to time looking into how to write a GitHub action).

Corrected the typos and formatted them to Black. Also added a PR to check if the code style is black. We can use another style as well but had the workflow lying around for black.

File: requirements.txt [hunk 6]
```diff
@@ -15,4 +15,7 @@ wasabi~=0.9.0
 pydantic~=1.8.2
 
 # testing
-pytest~=6.2.5
\ No newline at end of file
+pytest~=6.2.5
+
+# format
+black
```
Comment (saattrupdan) [inline requirements.txt:21]: Version?

**Review by saattrupdan — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 6 above)
Comment (KennethEnevoldsen) [reply requirements.txt:21]: Now added

**Comment by KennethEnevoldsen** [comment]
@HLasse will you take a look at this when you are back from vacation? Seems like there is an issue with loading DAGW from HF hub.

This has previously been an issue with the name change, but that does not seem to be the case now. An alternative might be that it is simply not up either on HF (which I doubt) or on DAGW (which I believe is more likely).

**Pull request merged**